Source/SuperLinq/Move.cs

```diff
@@ -72,4 +72,52 @@ static IEnumerable<T> Core(IEnumerable<T> source, int bufferStartIndex, int buff
 				yield return e.Current;
 		}
 	}
-}
+
+	/// <summary>
+	/// 	Returns a sequence with a range of elements in the source sequence moved to a new offset.
+	/// </summary>
+	/// <typeparam name="T">
+	/// 	Type of the source sequence.
+	/// </typeparam>
+	/// <param name="source">
+	/// 	The source sequence.
+	/// </param>
+	/// <param name="range">
+	/// 	The range of values to move.
+	/// </param>
+	/// <param name="toIndex">
+	/// 	The index where the specified range will be moved.</param>
+	/// <returns>
+	/// 	A sequence with the specified range moved to the new position.
+	/// </returns>
+	/// <exception cref="ArgumentNullException">
+	///	    <paramref name="source"/> is <see langword="null"/>.
+	/// </exception>
+	/// <exception cref="ArgumentOutOfRangeException">
+	///	    <paramref name="range"/>'s start is less than <c>0</c> or <paramref name="range"/>'s end is before start in the sequence.
+	/// </exception>
+	/// <remarks>
+	/// 	This operator uses deferred executing and streams its results.
+	/// </remarks>
+	public static IEnumerable<T> Move<T>(this IEnumerable<T> source, Range range, Index toIndex)
+	{
+		int? length = 0;
+		if (range.Start.IsFromEnd || range.End.IsFromEnd || toIndex.IsFromEnd)
+		{
+			length = source.TryGetCollectionCount();
+			if (!length.HasValue)
+			{
+				length = source.GetCollectionCount();
```

Comment:
if `TryGetCollectionCount()` fails to return a value, then `GetCollectionCount()` will throw an exception. try calling `Enumerable.Range(1, 10).Select(Identity).GetCollectionCount()`.